tests/integration/test_enrollment/conftest.py

```diff
@@ -0,0 +1,333 @@
+from wazuh_testing.tools.configuration import set_section_wazuh_conf, write_wazuh_conf
+from wazuh_testing.tools.monitoring import ManInTheMiddle
+from wazuh_testing.tools.security import CertificateController
+
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_conf.yaml')
+
+# Default data
+
+DEFAULT_VALUES = {
+    'enabled': 'yes',
+    'manager_address': '127.0.0.1',
+    'port': 1515,
+    'host_name': socket.gethostname(),
+    'groups': None,
+    'agent_address': '127.0.0.1',
+    'use_source_ip': 'no',
+    'ssl_cipher': None,
+    'server_ca_path': None,
+    'agent_certificate_path': None,
+    'agent_key_path': None,
+    'authorization_pass_path': None
+}
+
+folder = 'etc' if platform.system() == 'Linux' else ''
+CLIENT_KEYS_PATH = os.path.join(WAZUH_PATH, folder, 'client.keys')  # for unix add 'etc'
+AUTHDPASS_PATH = os.path.join(WAZUH_PATH, folder, 'authd.pass')
+SERVER_KEY_PATH = os.path.join(WAZUH_PATH, folder, 'manager.key')
+SERVER_CERT_PATH = os.path.join(WAZUH_PATH, folder, 'manager.cert')
+SERVER_PEM_PATH = os.path.join(WAZUH_PATH, folder, 'manager.pem')
+AGENT_KEY_PATH = os.path.join(WAZUH_PATH, folder, 'agent.key')
+AGENT_CERT_PATH = os.path.join(WAZUH_PATH, folder, 'agent.cert')
+AGENT_PEM_PATH = os.path.join(WAZUH_PATH, folder, 'agent.pem')
+AUTHDPASS_PATH = os.path.join(WAZUH_PATH, folder, 'authd.pass')
+AGENT_AUTH_BINARY_PATH = '/var/ossec/bin/agent-auth' if platform.system() == 'Linux' else \
+    os.path.join(WAZUH_PATH, 'agent-auth.exe')
+
+CONFIG_PATHS = {
+    'SERVER_PEM_PATH': SERVER_PEM_PATH,
+    'AGENT_CERT_PATH': AGENT_CERT_PATH,
+    'AGENT_PEM_PATH': AGENT_PEM_PATH,
+    'AGENT_KEY_PATH': AGENT_KEY_PATH,
+    'PASSWORD_PATH': AUTHDPASS_PATH
+}
+
+
+def parse_configuration_string(configuration):
+    """Formats a configuration dictionary with the default CONFIG_PATHS.
+    Args:
+        Configuration (dict): Configuration dictionary to be extended with CONFIG_PATHS.
+    """
+    for key, value in configuration.items():
+        if isinstance(value, str):
+            configuration[key] = value.format(**CONFIG_PATHS)
+
+
+# Test cases
+
+def load_tests(path):
+    """ Loads a yaml file from a path.
+    Args:
+        path (str): Location of the test to be loaded.
+    Returns:
+        yaml structure: Loaded test.
+    """
+    with open(path) as f:
+        return yaml.safe_load(f)
+
+
+tests = load_tests(os.path.join(test_data_path, 'wazuh_enrollment_tests.yaml'))
+
+CURRENT_TEST_CASE = {}
+
+
+@pytest.mark.parametrize('test_case', [case for case in tests])
+@pytest.fixture(scope="function")
+def set_test_case(test_case):
+    """Sets the current test case as a global variable accesible for different fixtures and callbacks.
+
+    Args: test_case (dict): The current test case.
+    """
+    global CURRENT_TEST_CASE
+    CURRENT_TEST_CASE = test_case
```

Comment:
The reason for setting the test case as a global variable is to reach it with the callback function **receiver_callback()**.
Let me know if there is a way to parameterize the callback function to avoid this approach.

Reply:
**get_configuration** loads the configuration of the test, the one from **wazuh_enrollment_conf.yaml**. Not the test_cases from **wazuh_enrollment_conf.yaml**. 
Do you refer to another approach for doing this?